src/plugins/data/common/search/search_source/search_source.test.ts

```diff
@@ -432,9 +432,9 @@ describe('SearchSource', () => {
         searchSource.setField('index', ({
           ...indexPattern,
           getComputedFields: () => ({
-            storedFields: [],
+            storedFields: ['*'],
             scriptFields: [],
-            docvalueFields: [],
+            docvalueFields: ['foo-mar'],
```

Comment:
It has admittedly been awhile since I looked at these tests, but I'm having a hard time understanding how these changes test the updates you've made to search source?

It would be nice to add a dedicated unit test asserting that metafields are specifically excluded from `fields`. That way if someone finds this in the future, they know it's intentional :)

Reply:
They don't, I was just trying to reproduce the scenario that was happening in the linked issue above. TBH, I don't understand how this was not caught by the unit test in the first place, because if I'm reading the test correctly, we never requested meta fields at all 🤷‍♀️

Reply:
> If I'm reading the test correctly, we never requested meta fields at all 🤷‍♀️

If the mocked index pattern includes the meta fields, then they will be sent in the request using the current test. Making this change causes the test to fail (and I expect it will pass with your changes here):

```diff
const indexPattern = ({
  title: 'foo',
- fields: [{ name: 'foo-bar' }, { name: 'field1' }, { name: 'field2' }],
+ fields: [{ name: 'foo-bar' }, { name: 'field1' }, { name: 'field2' }, { name: '_id' }],
  getComputedFields,
  getSourceFiltering: () => mockSource,
} as unknown) as IndexPattern;
```

So as long as `indexPattern.fields` contains those meta fields (which I assume it does?), then they'll be sent in the request. So it looks like metafields in the index pattern was a case which the existing tests didn't cover.

Reply:
aah, that's the thing I was missing. updated it now!